app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,46 @@
+class PostsController < ApplicationController
+
+  def index
+    @post = Post.all
+  end
+
+  def new
+    @post = Post.new
+  end
+
+
+  def create
+    @post = Post.create!(post_params)
+
+    @post.save
+    redirect_to @post
+  end
+
+  def show
+    @post = Post.find(params[:id])
+  end
+
+
+  def edit
+    @post = Post.find(params[:id])
+  end
+
+  def update
+    @post = Post.find(params[:id])
+    @postt.update(artist_params)
+
+    redirect_to posts_path(@post)
```

Comment:
will this work? you're redirecting to `posts_path`, which assumes multiple posts (Id guess the index). but you're passing in a single post